Code/Framework/AzCore/Tests/DOM/DomPrefixTreeTests.cpp

```diff
@@ -255,35 +255,67 @@ namespace AZ::Dom::Tests
         EXPECT_EQ(80, *tree.ValueAtPath(Path("/foo/0"), PrefixTreeMatch::ExactPath));
     }
 
-    TEST_F(DomPrefixTreeTests, AttachSubTreeAddsNewNode)
+    TEST_F(DomPrefixTreeTests, OverwritePathAddsNewNodeAtExistingPath)
     {
         DomPrefixTree<int> tree;
         tree.SetValue(Path("/foo"), 40);
 
-        DomPrefixTree<int> treeToAttach;
-        treeToAttach.SetValue(Path(), 80);
-        treeToAttach.SetValue(Path("/1"), 120);
-        tree.AttachSubTree(Path("/foo/0"), AZStd::move(treeToAttach));
+        DomPrefixTree<int> subtree;
+        subtree.SetValue(Path(), 80);
+        subtree.SetValue(Path("/1"), 120);
+        bool isOverwritten = tree.OverwritePath(Path("/foo/0"), AZStd::move(subtree));
 
-        // Validate that the new values is present at the paths in the original tree.
+        // Validate that the subtree is overwritten and new values are present at the paths in the original tree.
+        EXPECT_TRUE(isOverwritten);
```

Comment:
This is the use case I'm referring to. 'foo/0' didn't exist in the the tree before. So overwriting the path and returning true for this function would be confusing.